src/engine/QueryPlanner.cpp

```diff
@@ -1583,13 +1589,122 @@ std::vector<SubtreePlan> QueryPlanner::runGreedyPlanningOnConnectedComponent(
   return result;
 }
 
+// _____________________________________________________________________________
+std::vector<SubtreePlan> QueryPlanner::seedImplicitSpatialJoin(
+    std::vector<SparqlFilter> filters) {
+  std::vector<SubtreePlan> plans;
+
+  // TODO<ullingerc> Add support for more optimizable filters:
+  // * geof:distance() < constant
+  // * constant > geof:distance()
+  // * constant >= geof:distance()
+
+  // Helper lambda to detect and extract values from a comparison expression
+  // over geographic distance.
+  auto dissectWithinDistFilter =
+      [](const sparqlExpression::SparqlExpression& filterBody)
+      -> std::optional<std::pair<sparqlExpression::GeoFunctionCall, size_t>> {
+    // Supported comparison operator
+    auto compareExpr =
+        dynamic_cast<const sparqlExpression::LessEqualExpression*>(&filterBody);
+    if (compareExpr == nullptr) {
+      return std::nullopt;
+    }
+
+    // Left child must be distance function
+    auto children = compareExpr->children();
+    const auto& leftChild = *children[0];
+
+    // Right child must be constant
+    auto rightChild = children[1].get();
+    auto literalExpr =
+        dynamic_cast<sparqlExpression::detail::LiteralExpression<ValueId>*>(
+            rightChild);
+    if (literalExpr == nullptr) {
+      return std::nullopt;
+    }
+    ValueId constant = literalExpr->value();
+    // TODO<ullingerc> After merge of #1938 handle distance unit
+    size_t maxDist = 0;
+    if (constant.getDatatype() == Datatype::Double) {
+      maxDist = static_cast<size_t>(round(constant.getDouble() * 1000));
+    } else if (constant.getDatatype() == Datatype::Int) {
+      maxDist = constant.getInt() * 1000;
+    } else {
+      return std::nullopt;
```

Comment:
Conceptually this can be done in the `RelationalExpressions.cpp` file, 
so a function `getGeoFilter` that returns you
`?a, ?b, withinDist(27)`

Every code that is not inside `QueryPlanner.cpp` is good code.

Reply:
Also smaller functions are better to test.